### ISSUE TITLE
Support aarch64

### ISSUE THREAD

**Issue by thawkins** [JetBrains/pty4j#82]
Running Intelij Community on a raspberry PI model 4 (4G) has a problem with loading pty4j and subsequently cant open terminals in the interface. This seems to be due to a lack of support for ARM 64 architecture. Intelij is fast and highly useable on Raspbian on pi4, I dont think it would be nessacary to support earlier PIs as pi4 is the first version that has suffient ram and performance to run Intelij. 

CPU info is:
```
pi@raspberrypi:~/tmp $ more /proc/cpuinfo
processor	: 0
model name	: ARMv7 Processor rev 3 (v7l)
BogoMIPS	: 108.00
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt 
vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

processor	: 1
model name	: ARMv7 Processor rev 3 (v7l)
BogoMIPS	: 108.00
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt 
vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

processor	: 2
model name	: ARMv7 Processor rev 3 (v7l)
BogoMIPS	: 108.00
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt 
vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

processor	: 3
model name	: ARMv7 Processor rev 3 (v7l)
BogoMIPS	: 108.00
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt 
vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

Hardware	: BCM2835
Revision	: c03111
Serial		: 10000000fade59cd
Model		: Raspberry Pi 4 Model B Rev 1.1
pi@raspberrypi:~/tmp $ 
```

kernel info is:
```
pi@raspberrypi:~/tmp $ uname -a
Linux raspberrypi 4.19.75-v7l+ #1270 SMP Tue Sep 24 18:51:41 BST 2019 armv7l GNU/Linux
pi@raspberrypi:~/tmp $ 

```
idea.log showing loading issue. 
```
2019-12-07 10:06:56,788 [1682070]   INFO - erminal.AbstractTerminalRunner - Cannot open Local Terminal 
java.util.concurrent.ExecutionException: Failed to start [/bin/bash, --rcfile, /home/pi/bin/intelij/idea-IC-193.5233.102/plugins/terminal/jediterm-bash.in, -i] in /home/pi/IdeaProjects/ktest
	at org.jetbrains.plugins.terminal.LocalTerminalDirectRunner.createProcess(LocalTerminalDirectRunner.java:197)
	at org.jetbrains.plugins.terminal.LocalTerminalDirectRunner.createProcess(LocalTerminalDirectRunner.java:47)
	at org.jetbrains.plugins.terminal.AbstractTerminalRunner.lambda$openSessionInDirectory$4(AbstractTerminalRunner.java:214)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:238)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Exec_tty error:Cannot load native pty executor library
	at com.pty4j.unix.UnixPtyProcess.execInPty(UnixPtyProcess.java:272)
	at com.pty4j.unix.UnixPtyProcess.<init>(UnixPtyProcess.java:86)
	at com.pty4j.PtyProcessBuilder.start(PtyProcessBuilder.java:111)
	at com.pty4j.PtyProcess.exec(PtyProcess.java:78)
	at com.pty4j.PtyProcess.exec(PtyProcess.java:49)
	at org.jetbrains.plugins.terminal.LocalTerminalDirectRunner.createProcess(LocalTerminalDirectRunner.java:189)
	... 8 more
Caused by: java.lang.RuntimeException: Cannot load native pty executor library
	at com.pty4j.unix.PtyHelpers.getPtyExecutor(PtyHelpers.java:257)
	at com.pty4j.unix.PtyHelpers.execPty(PtyHelpers.java:430)
	at com.pty4j.unix.UnixPtyProcess.exec(UnixPtyProcess.java:333)
	at com.pty4j.unix.UnixPtyProcess$Reaper.execute(UnixPtyProcess.java:389)
	at com.pty4j.unix.UnixPtyProcess$Reaper.run(UnixPtyProcess.java:395)
Caused by: java.lang.UnsatisfiedLinkError: Unable to load library '/home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so':
/home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so: cannot open shared object file: No such file or directory
/home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so: cannot open shared object file: No such file or directory
Native library (home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so) not found in resource path (/home/pi/bin/intelij/idea-IC-193.5233.102/lib/bootstrap.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/extensions.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/util.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/jdom.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/log4j.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/trove4j.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/jna.jar:/usr/lib/jvm/java-8-openjdk-armhf/lib/tools.jar)
	at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:302)
	at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:455)
	at com.sun.jna.Library$Handler.<init>(Library.java:192)
	at com.sun.jna.Native.loadLibrary(Native.java:646)
	at com.sun.jna.Native.loadLibrary(Native.java:630)
	at com.pty4j.unix.NativePtyExecutor.<init>(NativePtyExecutor.java:14)
	at com.pty4j.unix.PtyHelpers$2.call(PtyHelpers.java:222)
	at com.pty4j.unix.PtyHelpers$2.call(PtyHelpers.java:218)
	at com.pty4j.util.LazyValue.getValue(LazyValue.java:25)
	at com.pty4j.unix.PtyHelpers.getPtyExecutor(PtyHelpers.java:254)
	at com.pty4j.unix.PtyHelpers.<clinit>(PtyHelpers.java:234)
	at com.pty4j.unix.Pty.ptyMasterOpen(Pty.java:140)
	at com.pty4j.unix.Pty.openMaster(Pty.java:176)
	at com.pty4j.unix.Pty.<init>(Pty.java:55)
	at com.pty4j.unix.UnixPtyProcess.<init>(UnixPtyProcess.java:83)
	at com.pty4j.PtyProcessBuilder.start(PtyProcessBuilder.java:111)
	at com.pty4j.PtyProcess.exec(PtyProcess.java:78)
	at com.pty4j.PtyProcess.exec(PtyProcess.java:49)
	at org.jetbrains.plugins.terminal.LocalTerminalDirectRunner.createProcess(LocalTerminalDirectRunner.java:189)
	at org.jetbrains.plugins.terminal.LocalTerminalDirectRunner.createProcess(LocalTerminalDirectRunner.java:47)
	at org.jetbrains.plugins.terminal.AbstractTerminalRunner.lambda$openSessionInDirectory$4(AbstractTerminalRunner.java:214)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:238)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	Suppressed: java.lang.UnsatisfiedLinkError: /home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so: cannot open shared object file: No such file or directory
		at com.sun.jna.Native.open(Native Method)
		at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:191)
		... 26 more
	Suppressed: java.lang.UnsatisfiedLinkError: /home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so: cannot open shared object file: No such file or directory
		at com.sun.jna.Native.open(Native Method)
		at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:204)
		... 26 more
	Suppressed: java.io.IOException: Native library (home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so) not found in resource path (/home/pi/bin/intelij/idea-IC-193.5233.102/lib/bootstrap.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/extensions.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/util.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/jdom.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/log4j.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/trove4j.jar:/home/pi/bin/intelij/idea-IC-193.5233.102/lib/jna.jar:/usr/lib/jvm/java-8-openjdk-armhf/lib/tools.jar)
		at com.sun.jna.Native.extractFromResourcePath(Native.java:1095)
		at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:276)
		... 26 more
2019-12-07 10:07:14,600 [1699882]   INFO - rationStore.ComponentStoreImpl - Saving Module: 'ktest.linuxMain'FacetManager took 32 ms 
2019-12-07 10:07:14,643 [1699925]   INFO - rationStore.ComponentStoreImpl - Saving Module: 'ktest.linuxTest'FacetManager took 41 ms 
```

**Comment by RockyBoulder** [comment]
Same problem here using PyCharm on a rock64  which is aarch64, too.

**Comment by lezsakdomi** [comment]
Same problem here using IDEA on a Pinebook and a Pinebook Pro, also aarch64.

The `/home/led/Downloads/idea-IU-193.5662.53/lib/pty4j-native/linux/x86/libpty.so` , but in the error log there is `home/led/Downloads/idea-IU-193.5662.53/lib/pty4j-native/linux/x86/libpty.so` (without the starting slash). I did not try creating a symlink to `/home` as `home`.

My full log can be read at https://termbin.com/4gbpf

**Comment by aleroot** [comment]
In order to solve the problem you can simply compile the library yourself, for example for aarch4: 

- pty4j/native
- gcc -fPIC -c *.c
- gcc -shared -o libpty.so *.o
- cp libpty.so /opt/idea-IC/lib/pty4j-native/linux/x86_64/libpty.so

Note that IntelliJ look for the library in x86_64 folder even if we are under aarch4 :)

**Comment by tuomastik** [comment]
> In order to solve the problem you can simply compile the library yourself, for example for aarch4:
> 
> * pty4j/native
> * gcc -fPIC -c *.c
> * gcc -shared -o libpty.so *.o
> * cp libpty.so /opt/idea-IC/lib/pty4j-native/linux/x86_64/libpty.so
> 
> Note that IntelliJ look for the library in x86_64 folder even if we are under aarch4 :)

This fixed the error also for PyCharm 2020.1 running on Raspberry Pi 4. Just to clarify the compiling steps: the `gcc` commands need to be run in directory `pty4j/native` and in PyCharm's case `libpty.so` needs to be copied under `pycharm-community-2020.1/lib/pty4j-native/linux/x86/libpty.so`

**Comment by ClarenceChen96** [comment]
I am a new programmer. I am still confused about the solution above. I have the same problem, but  seems my '/home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so' file is present, and when I tried the gcc commands in 'home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/' it shows 
gcc: error: */c: No such file or directory
gcc: fatal error: no input files
compilation terminated.

**Comment by tuomastik** [comment]
> I am a new programmer. I am still confused about the solution above. I have the same problem, but seems my '/home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so' file is present, and when I tried the gcc commands in 'home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/' it shows
> gcc: error: */c: No such file or directory
> gcc: fatal error: no input files
> compilation terminated.

Clone **this repository** and run the `gcc` commands under its `native` directory.

**Comment by ClarenceChen96** [comment]
Thank you for your clarification! I misunderstood earlier. Solved my problem.

**Comment by Morgan77777** [comment]
I am also new to programming and struggling to understand the fix for PyCharm on the Raspberry Pi4. 

> > I am a new programmer. I am still confused about the solution above. I have the same problem, but seems my '/home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so' file is present, and when I tried the gcc commands in 'home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/' it shows
> > gcc: error: */c: No such file or directory
> > gcc: fatal error: no input files
> > compilation terminated.
> 
> Clone **this repository** and run the `gcc` commands under its `native` directory.

What is this repository referring to and what would the solution be for the same error messages PyCharm on the Pi 4?

**Comment by tuomastik** [comment]
> I am also new to programming and struggling to understand the fix for PyCharm on the Raspberry Pi4.
> 
> > > I am a new programmer. I am still confused about the solution above. I have the same problem, but seems my '/home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so' file is present, and when I tried the gcc commands in 'home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/' it shows
> > > gcc: error: */c: No such file or directory
> > > gcc: fatal error: no input files
> > > compilation terminated.
> > 
> > 
> > Clone **this repository** and run the `gcc` commands under its `native` directory.
> 
> What is this repository referring to and what would the solution be for the same error messages with PyCharm on the Pi 4?

```bash
git clone https://github.com/JetBrains/pty4j.git
cd pty4j/native
gcc -fPIC -c *.c
gcc -shared -o libpty.so *.o
cp libpty.so YOUR_PYCHARM_INSTALL_LOCATION/lib/pty4j-native/linux/x86/
```

**Comment by RafaelZasas** [comment]
I did everything specified and no errors were thrown at me but no problems were solved either...

**Comment by mrdschrute** [comment]
> > I am also new to programming and struggling to understand the fix for PyCharm on the Raspberry Pi4.
> > > > I am a new programmer. I am still confused about the solution above. I have the same problem, but seems my '/home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/linux/x86/libpty.so' file is present, and when I tried the gcc commands in 'home/pi/bin/intelij/idea-IC-193.5233.102/lib/pty4j-native/' it shows
> > > > gcc: error: */c: No such file or directory
> > > > gcc: fatal error: no input files
> > > > compilation terminated.
> > > 
> > > 
> > > Clone **this repository** and run the `gcc` commands under its `native` directory.
> > 
> > 
> > What is this repository referring to and what would the solution be for the same error messages with PyCharm on the Pi 4?
> 
> ```shell
> git clone https://github.com/JetBrains/pty4j.git
> cd pty4j/native
> gcc -fPIC -c *.c
> gcc -shared -o libpty.so *.o
> cp libpty.so YOUR_PYCHARM_INSTALL_LOCATION/lib/pty4j-native/linux/x86/
> ```
Was having this issue on a pi4 running itellij. This fixed it. Thank you.

**Comment by JensenHJS** [comment]
thank you so much
我用此方法解决了nvidia jetson tx2上clion无法使用terminal的问题，此方法应该也适用于pycharm和aarch64的所有硬件平台。thank you so much

git clone https://github.com/JetBrains/pty4j.git
cd pty4j/native
gcc -fPIC -c *.c
gcc -shared -o libpty.so *.o
cp libpty.so YOUR_PYCHARM_INSTALL_LOCATION/lib/pty4j-native/linux/x86/

**Comment by ikostan** [comment]
> Thank you for your clarification! I misunderstood earlier. Solved my problem.
> In my case, java was looking into the 'x86' folder instead of 'x86_64'. Had to replace it in the commands above.

Hi all.

 

My setup:

    Manjaro ARM Linux (Kernel Linux 5.4.51-2-MANJARO-ARM (aarch64))
    Raspberry Pi 4 4GB
    PyCharm 2020.2 (Community Edition)

I had same issue. Solved it by running the last step as following:

`cp libpty.so /opt/pycharm-community-2020.2/lib/pty4j-native/linux/x86_64/`

**Comment by davidacarroll** [comment]
I've found this post since I am running into the same problem on my Raspberry Pi 4 running Raspberry Pi OS 2020-08-20. I followed the steps above, copied the libpty.so file into my YOUR_PYCHARM_INSTALL_LOCATION/lib/pty4j-native/linux/x86/. However, when I compile I continue to get the same errors. Is there an extra step for me to set up my "resource path". It shows a list of .jar files it is checking. I am not sure how to resolve this part. Thanks

	Suppressed: java.io.IOException: Native library (opt/pycharm-community-2020.2.3/lib/pty4j-native/linux/x86/libpty.so) not found in resource path (/opt/pycharm-community-2020.2.3/lib/bootstrap.jar:/opt/pycharm-community-2020.2.3/lib/extensions.jar:/opt/pycharm-community-2020.2.3/lib/util.jar:/opt/pycharm-community-2020.2.3/lib/jdom.jar:/opt/pycharm-community-2020.2.3/lib/log4j.jar:/opt/pycharm-community-2020.2.3/lib/trove4j.jar:/opt/pycharm-community-2020.2.3/lib/jna.jar)
		at com.sun.jna.Native.extractFromResourcePath(Native.java:1095)
		at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:275)
		... 31 more

**Comment by segrey** [comment]
@davidacarroll Please provide output of `cat /proc/cpuinfo` (info for one of processors is enough). As a workaround, you might also need to copy libpty.so to `YOUR_PYCHARM_INSTALL_LOCATION/lib/pty4j-native/linux/x86_64`.

**Comment by davidacarroll** [comment]
Hello, and thanks for responding.

I did try moving libpty.so into
YOUR_PYCHARM_INSTALL_LOCATION/lib/pty4j-native/linux/x86_64
but had the same results.

Below is the output of /proc/cpuinfo:

pi@raspberrypi:~ $ cat /proc/cpuinfo
processor : 0
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt
vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

processor : 1
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt
vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

processor : 2
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt
vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

processor : 3
model name : ARMv7 Processor rev 3 (v7l)
BogoMIPS : 108.00
Features : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt
vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xd08
CPU revision : 3

Hardware : BCM2711
Revision : c03112
Serial : 10000000e6f113cc
Model : Raspberry Pi 4 Model B Rev 1.2
pi@raspberrypi:~ $
*David A Carroll*
Software Engineering | Operational Excellence | Leader | Customer Success
✉ davidacarroll.one@gmail.com
[image: Image result for linkedin icon]
<https://www.linkedin.com/in/david-carroll-one/>
https://www.linkedin.com/in/david-carroll-one/
☎ (919) 621-1270

**Comment by davidacarroll** [comment]
Ok, I tried again and copied the file to the x86_64 folder and it worked.
The prior time I moved the file. So I think I am all set for now. Thank you
very much!

*David A Carroll*
Software Engineering | Operational Excellence | Leader | Customer Success
✉ davidacarroll.one@gmail.com
[image: Image result for linkedin icon]
<https://www.linkedin.com/in/david-carroll-one/>
https://www.linkedin.com/in/david-carroll-one/
☎ (919) 621-1270

**Comment by segrey** [comment]
@davidacarroll Would be great if you could attach also output of `lscpu` and `uname -a`.

**Comment by FrederikBer** [comment]
Had the same Issue and tried all of the above, In the end I had to change 
`cp libpty.so [PyCharmInstallationFolder]/lib/pty4j-native/linux/x86/` to 
`cp libpty.so [PyCharmInstallationFolder]/lib/pty4j-native/linux/arm/`

I found this out by looking at the log where it said `ERROR -      com.pty4j.unix.PtyHelpers - Cannot load native pty executor library 
java.lang.UnsatisfiedLinkError: Unable to load library '/opt/pycharm-2020.3.4/lib/pty4j-native/linux/arm/libpty.so':
/opt/pycharm-2020.3.4/lib/pty4j-native/linux/arm/libpty.so: cannot open shared object file: No such file or directory`
Since the path and file were not present it obviously wouldnt work, after creating the `arm` directory and then using the changed command it worked. Just for anyone who is still searching for a solution.

**Comment by krystofcelba** [comment]
Hey,
I had the same issue but on macOS with an M1 processor and Android Studio Preview.
The above solution works but I had to change the commands slightly. I'm sharing it here for anyone with the same issue on M1 Mac :-). 

```sh
git clone https://github.com/JetBrains/pty4j.git
cd pty4j/native
clang -fPIC -c *.c
clang -shared -o libpty.dylib *.o
cp libpty.dylib "/Applications/Android Studio Preview.app/Contents/lib/pty4j-native/darwin/"
```

**Comment by segrey** [comment]
@krystofcelba M1 processors were supported in 4fd4f84d701f3418539c59322f5ae674e4e92cec (included in pty4j version 0.11.3 or higher). The pty4j version with the fix is also included in IntelliJ IDEs 2020.3.1 release. Please specify your Android Studio Preview version and version of bundled pty4j.

**Comment by krystofcelba** [comment]
@segrey It's 
```Android Studio Arctic Fox | 2020.3.1 Canary 15
Build #AI-203.7717.56.2031.7321754, built on April 29, 2021
Runtime version: 11.0.10+0-b96-7249189 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 11.3
GC: G1 Young Generation, G1 Old Generation
Memory: 1920M
Cores: 8
Registry: external.system.auto.import.disabled=true
```
And pty4j `pty4j-0.11.1.jar` (so lower)

It should be the latest Android Studio with M1 support I think.

**Comment by segrey** [comment]
@krystofcelba Awesome, thanks for the details! I guess the next Android Studio release will have the fixed version of pty4j.

**Comment by hunterseeker** [comment]
> > In order to solve the problem you can simply compile the library yourself, for example for aarch4:
> > 
> > * pty4j/native
> > * gcc -fPIC -c *.c
> > * gcc -shared -o libpty.so *.o
> > * cp libpty.so /opt/idea-IC/lib/pty4j-native/linux/x86_64/libpty.so
> > 
> > Note that IntelliJ look for the library in x86_64 folder even if we are under aarch4 :)
> 
> This fixed the error also for PyCharm 2020.1 running on Raspberry Pi 4. Just to clarify the compiling steps: the `gcc` commands need to be run in directory `pty4j/native` and in PyCharm's case `libpty.so` needs to be copied under `pycharm-community-2020.1/lib/pty4j-native/linux/x86/`

Thanks to all on this post for help on this.  After I executed the recommended commands (above), I ran pycharm.sh (my shortcut on desktop on RaspPi4), and chose ''Execute in terminal' option.  I examined what was reported in terminal, and found that I needed to create an 'arm' folder in the linux folder mentioned above.  So instead of x86 or x86_64, the folder 'arm' was required. Copied the libpty.so to the new arm folder, and it worked.

**Comment by segrey** [comment]
@hunterseeker It should be fixed in the next PyCharm major release - 2021.2 (https://github.com/JetBrains/pty4j/issues/93).
